app/src/main/java/com/humberto/tasky/core/database/mapper/EventMappers.kt

```diff
@@ -0,0 +1,76 @@
+package com.humberto.tasky.core.database.mapper
+
+import com.humberto.tasky.core.database.entity.AttendeeEntity
+import com.humberto.tasky.core.database.entity.EventEntity
+import com.humberto.tasky.core.database.entity.PhotoEntity
+import com.humberto.tasky.core.domain.event.Attendee
+import com.humberto.tasky.core.domain.event.Event
+import com.humberto.tasky.core.domain.event.Photo
+import java.util.UUID
+
+fun EventEntity.toEvent(
+    getAttendeesEntitiesByIds: (List<String>) -> List<AttendeeEntity>,
+    getPhotosEntitiesByUrls: (List<String>) -> List<PhotoEntity>
```

Comment:
Fetching these entities will most likely involve the local database and those calls you ideally want to make in a suspending function to not block the main thread. If you had to change these functions to be suspending, you’d be in a situation where you can only map an entity to an Event when in a coroutine, which is not ideal.

Do you think it could make more sense to update your `EventEntity` to include the photos and attendee objects?

Reply:
Alternatively, making this an `inline` function would make sense. You could then call it in and outside of a coroutine